### DISCUSSION TITLE
How to install the library + header files?

### DISCUSSION THREAD

**Discussion by troglobit** [EIPStackGroup/OpENer#373] [Q&A] [answered]
Hi, first time user here!

I've successfully built and run the `bin/posix/src/ports/POSIX/OpENer` program, which seems to work fine with my scanner.

Now I'm curious how to create my own application, based on the `sample_application`, in my own git repo.  _So I want to install the libraries and header files_, e.g. in `/usr/local`, but I can't wrap my head around _how to do that?_  

Admittedly I'm a noob also with cmake ...

**Comment by troglobit** [comment]
So, there's something afoul with `bin/posix/setup_posix.sh`, dunno why but I can't get `sudo make install` to work with that. The install target is missing?

If otoh I configure with the following:
```
mkdir -p build
cd build/
cmake	-DCMAKE_C_COMPILER=gcc						\
	-DCMAKE_INSTALL_PREFIX:PATH="/usr/local/opener" 		\
	-DOpENer_PLATFORM:STRING="POSIX"				\
	-DOpENer_Device_Config_Vendor_Id:STRING="${VENDOR_ID}"		\
	-DOpENer_Device_Config_Device_Name:STRING="${DEVICE_NAME}"	\
	-DOpENer_Device_Config_Device_Type:STRING="${DEVICE_TYPE}"	\
	-DOpENer_Device_Config_Product_Code:STRING="${PRODUCT_CODE}"	\
	-DOPENER_BUILD_SHARED_LIBS=ON					\
	../source
```
Then I can do `sudo make install`, but there are no .so file(s) built or installed.  Any cmake ninja out there that can have a look?  I'm more of an autotools person myself so I'm riding rusty old my bike in fairly deep and unfamiliar waters here :smile:

**Comment by MartinMelikMerkumians** [comment]
Hi,
there is a CMake Flag called OPENER_BUILD_SHARED_LIBS which should create library files (basically everything without the sample_application afaik). I am usually not using OpENer in this mode, but it was provided here https://github.com/EIPStackGroup/OpENer/pull/184

**Reply by troglobit** [reply]
Hi!  Yeah I tried enabling that in my last follow-up in this discussion, see result above.

**Reply by MartinMelikMerkumians** [reply]
Hi.

I am also not a CMake guru (not sure if anyone really ist 😄) but there is no install target currently, as you suspected.
Not sure how much effort this is, I would have to check the documentation first (https://cmake.org/cmake/help/latest/command/install.html)

**Reply by troglobit** [reply]
Aha, OK I'll have a look at it later to see if I can patch something together.  Thanks! :smiley::+1:

**Comment by troglobit** [comment, marked as answer]
Turns out the problems were caused by a couple of regressions on the master branch.  Proposed fixes in PR #375

**Reply by MartinMelikMerkumians** [reply]
Thank you. Merged
